sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/certificate/_patch.py

```diff
@@ -0,0 +1,46 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+from typing import Any, List, Optional
+
+from azure.confidentialledger.certificate._client import (
+    ConfidentialLedgerCertificateClient as GeneratedClient,
+)
+
+__all__: List[str] = [
+    "ConfidentialLedgerCertificateClient"
+]  # Add all objects you want publicly available to users at this package level
+
+
+def patch_sdk():
+    """Do not remove from this file.
+
+    `patch_sdk` is a last resort escape hatch that allows you to do customizations
+    you can't accomplish using the techniques described in
+    https://aka.ms/azsdk/python/dpcodegen/python/customize
+    """
+
+
+class ConfidentialLedgerCertificateClient(GeneratedClient):
+    def __init__(  # pylint: disable=missing-client-constructor-parameter-credential
+        self, certificate_endpoint: Optional[str] = None, **kwargs: Any
+    ) -> None:
+        """
+        :param certificate_endpoint: The Identity Service URL, for example
+            https://identity.confidential-ledger.core.azure.com, defaults to None. If not provided,
+            "https://identity.confidential-ledger.core.azure.com" will be used.
```

Comment:
This endpoint was made optional across all languages?

Reply:
It is not optional in all languages, but it is constant so not requiring it seems to be convenient for the user. The closest example is in .NET, where the parameter on the `ConfidentialLedgerCertificateClient` is not optional, but if the `ConfidentialLedgerClient` does auto-magic to retrieve the Ledger's TLS certificate, it uses a hardcoded value for the `certificateEndpoint`.

Reply:
I'm not sure I would make this optional... while we can do so later without it being breaking, the reverse is not possible.
@johanste - do you have thoughts on this?

Reply:
Assuming that the value provided by default will continue to a valid option for at least some customers, I don't see a major issue here. Unless we are fairly certain that in the future, the the default will be incorrect, for the majority of users, having a default value and allow people to override seem reasonable. I don't have a strong opinion on this, though - it is going to depend on how often we expect the default to be co

Reply:
We don't expect the endpoint to change at all now, especially since a change would require a lot of certificate reprovisioning and so on.